File: android/ffmpeg-kit-android-lib/src/main/java/com/lgcns/ffmpegkit/ReturnCode.java
package com.lgcns.ffmpegkit;

public class ReturnCode {

    public static int SUCCESS = 0;

    public static int CANCEL = 255;

    private final int value;

    public ReturnCode(final int value) {
        this.value = value;
    }

    public static boolean isSuccess(final ReturnCode returnCode) {
        return (returnCode != null && returnCode.getValue() == SUCCESS);
    }

    public static boolean isCancel(final ReturnCode returnCode) {
        return (returnCode != null && returnCode.getValue() == CANCEL);
    }

    public int getValue() {
        return value;
    }

    public boolean isValueSuccess() {
        return (value == SUCCESS);
    }

    public boolean isValueError() {
        return ((value != SUCCESS) && (value != CANCEL));
    }

    public boolean isValueCancel() {
        return (value == CANCEL);
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }

}

File: android/ffmpeg-kit-android-lib/src/main/java/com/lgcns/ffmpegkit/ReturnCode.kt
package com.lgcns.ffmpegkit

class ReturnCode(@JvmField val value: Int) {

    val isValueSuccess: Boolean
        get() = value == SUCCESS
    val isValueError: Boolean
        get() = value != SUCCESS && value != CANCEL
    val isValueCancel: Boolean
        get() = value == CANCEL

    override fun toString(): String {
        return value.toString()
    }

    companion object {
        @JvmField
        var SUCCESS = 0
        @JvmField
        var CANCEL = 255
        fun isSuccess(returnCode: ReturnCode?): Boolean {
            return returnCode != null && returnCode.value == SUCCESS
        }

        fun isCancel(returnCode: ReturnCode?): Boolean {
            return returnCode != null && returnCode.value == CANCEL
        }
    }
}